src/CodeGen_Hexagon.cpp

```diff
@@ -2098,6 +2110,13 @@ void CodeGen_Hexagon::visit(const Select *op) {
         value = codegen(Call::make(op->type, Call::if_then_else,
                                    {b->value, op->true_value, op->false_value},
                                    Call::PureIntrinsic));
+    } else if (op->type.is_vector() && op->type.is_bool()) {
+        // Lower selects on bools to bit math
+        std::string cond_name = unique_name('c');
+        Expr cond = Variable::make(op->condition.type(), cond_name);
+        Expr equiv = Let::make(cond_name, op->condition,
+                               (op->true_value & cond) | (op->false_value & ~cond));
+        value = codegen(equiv);
```

Comment:
This seems out of scope for the PR. Does the rest of the PR break something that this fixes?

Reply:
Oh I missed the handful of select-outputting rules.